### PR TITLE
Make the output of versioninfo() customizable with kwargs

### LINKED ISSUE #21971: include values of significant environment vars in versioninfo()
See e.g. https://github.com/JuliaLang/julia/issues/21958: the output of `versioninfo()` should include the values of any `JULIA_` environment variables that can affect the behavior of the program.

### PR DESCRIPTION
**Pull request by fredrikekre**
also reordered some things in a more logical order. Fix #21971 

What kwargs should we have?

### CODE REVIEW

File: base/interactiveutil.jl [hunk 1]
```diff
@@ -274,42 +274,61 @@ function versioninfo(io::IO=STDOUT, verbose::Bool=false)
...
-        println(io,         "  BLAS: ",libblas_name)
+        println(io, "  BLAS: ",libblas_name)
+    end
+    println(io, "  LAPACK: ",liblapack_name)
+    println(io, "  LIBM: ",libm_name)
+    println(io, "  LLVM: libLLVM-",libllvm_version," (", Sys.JIT, ", ", Sys.cpu_name, ")")
+
+    println(io, "Environment:")
+    for (k,v) in ENV
+        if ismatch(r"JULIA", String(k))
```
Comment (StefanKarpinski) [inline base/interactiveutil.jl]: Should we look for prefixes only? Include `JL_`-prefixed variables too? We've tried to move away from those, but there may be a few lingering here or there.

Comment (fredrikekre) [inline base/interactiveutil.jl]: I simply hoisted the printing of `JULIA_*` variables to always be displayed and the rest of the env vars for `verbose=true`. Add `JL_ `prefixed env vars to always show too?

Comment (StefanKarpinski) [inline base/interactiveutil.jl]: Nah, don't bother – I grepped the source code and it seems like we don't have any lingering environment variables that start with `JL_` anymore :)

**Comment by amitmurthy** [comment]
As part of this PR, could we also print the Linux kernel version? Currently we just display `OS: Linux (x86_64-linux-gnu)`. Will be good if it can display `OS: Linux (x86_64-linux-gnu), Kernel xx.xx.xx` 

The `uname` system call appears to provide this information - https://linux.die.net/man/2/uname

**Comment by fredrikekre** [comment]
Output of `uname` is included already. Do you propose to display it as default (it's currently only shown with `verbose=true`)?

**Comment by amitmurthy** [comment]
That is fine then. I was not aware of the `verbose` option.

**Comment by fredrikekre** [comment]
Rebased. I am happy with the PR as is. It is easy to add more kwargs later if need be.

**Comment by fredrikekre** [comment]
Rebased.

**Review by StefanKarpinski — APPROVED** [review]

**Pull request merged**